pkg/datapath/tables/route.go

```diff
@@ -53,12 +54,12 @@ type RouteID struct {
 }
 
 func (id RouteID) Key() index.Key {
-	key := append(index.Uint64(uint64(id.Table)), '+')
-	key = append(key, index.Uint64(uint64(id.Table))...)
-	key = append(key, '+')
-	key = append(key, index.NetIPPrefix(id.Dst)...)
-	key = append(key, 0 /* termination */)
-	return key
+	key := make([]byte, 0, 4 /* table */ +4 /* link */ +17 /* prefix & bits */)
+	key = binary.BigEndian.AppendUint32(key, uint32(id.Table))
+	key = binary.BigEndian.AppendUint32(key, uint32(id.LinkIndex))
+	addrBytes := id.Dst.Addr().As16()
```

Comment:
This return ipv4 mapped ipv6, I'm assuming this is OK for just creating the key? Just double checking.

Reply:
Uhm, that is actually a good point. I can imagine that one might have 2 routes an IPv4 and a IPv4-mapped-IPv6 of the same IP, which would conflict. But I don't know if IPv4 and IPv6 routes would be in the same table :thinking:. @joamaki would likely know, but he is busy. I will add an additional byte for IP family just to be sure, we can remove it later if it turns out to not be nessesery.

Reply:
Nevermind, Jussi already thought of this, after the IP he includes the bits count as 1 byte 4 or 16, so all good.

Reply:
Additional byte for uniqueness sounds good. 

Do you need a consistent size key ?  Because I imagine also calling "as4" when you have ipv4 would retain uniqueness. 

But either way, family byte works well.

Reply:
Got it! I sent that last comment before gh showed me you responded :).